module2/gift-certificate-system/service/src/main/java/com/epam/esm/service/TagServiceImpl.java

```diff
@@ -33,7 +34,12 @@ public TagServiceImpl(TagDao tagDao) {
 
     @Override
     public Tag findTagById(Long id) {
-        return tagDao.find(id).orElseThrow(TagNotFoundException::new);
+        Optional<Tag> tag = tagDao.find(id);
+        if (!tag.isPresent()) {
+            throw new TagNotFoundException("The tag not found");
+        } else {
+            return tag.get();
+        }
```

Comment:
It will be better to use smth like this:

Tag tag = tagDao.find(id)
    .orElseThrow(() -> new TagNotFoundException("..."));

Reply:
And in the other places as well